casanovo/denovo/model_runner.py

```diff
@@ -59,20 +70,39 @@ def __init__(
         self.loaders = None
         self.writer = None
 
-        best_filename = "best"
-        if output_rootname is not None:
-            best_filename = f"{output_rootname}.{best_filename}"
+        if output_dir is None:
+            self.callbacks = []
+            warnings.warn(
```

Comment:
**question:** Any benefit over emitting a warning vs logging with the warning level. I think the latter seems more relevant here.